### ISSUE TITLE
should a data-access lib with a facade export more than just the facade?

### ISSUE THREAD

**Issue by markgoho** [nrwl/nx#1085]
Currently, when running `ng g ngrx` with a facade generated as well, the schematic exports the facade, the reducer, and the selectors from the lib's index.ts file.

This is partially a best practice question, but one that would have an impact on the schematic.

The question is, for a library that's got an ngrx store in it and that's its only purpose, _and_ it has a facade, should we really be exporting the reducer and selectors as well? I believe the facade should be the only publicly available entry point to the lib.

This does, however, present some questions:
1. Selectors: If we want to compose selectors in another data-access lib using our original one, only having access to the facade makes this impossible.
2. Effects: If we have, say, a notifications lib that uses ngrx to dispatch global notifications, and we'd like to have the `new ShowNotification('Hello!')` action be dispatched in other Effects modules, we'd have to use the Facade method in a `tap` operator instead of returning a new action. This doesn't feel wrong, but it doesn't feel great either.

So, to summarize, should the public API of a data access lib be limited to _just_ the facade?

**Comment by vsavkin** [comment]
@ThomasBurleson do you have any thoughts on this?

**Comment by brandonroberts** [comment]
@markgoho the public API should not be limited to just the facade. Selectors and actions are meant to be shared across different areas that need to consume them.

1. Selectors still need to be exported because you should combine shared data in your Store using selectors, and not with Facades + Observable selectors + combineLatest. Using the pattern with a shared data source creates unnecessary computations when multiple data sets are changed at the same time.
2. You should be mapping to actions in Effects and not using `tap()` from facades.

Facades help you isolate NgRx from your components, but past that you're going to run into these issues where you're trying to combine two patterns that don't mesh well in certain use cases.
